src/Equations/Fluid/HottailRateTermLowZ.cpp

```diff
@@ -0,0 +1,423 @@
+    id_ni(unknowns->GetUnknownID(OptionConstants::UQTY_ION_SPECIES)) // Needed for derivatives in jacobian
+{
+    SetName("HottailRateTermLowZ");
+
+    //AddUnknownForJacobian(unknowns,id_Efield);
+    AddUnknownForJacobian(unknowns,id_ncold);
+    AddUnknownForJacobian(unknowns,id_tau);
+    AddUnknownForJacobian(unknowns,id_ni); //Also affects Zeff, now we neglect this contribution. 
+    AddUnknownForJacobian(unknowns,id_Tcold);
+    
+    this->w = gsl_integration_workspace_alloc(1000);
+    this->rGrid = grid->GetRadialGrid();
+    
+    this->GridRebuilt();
+}
+
+/**
+ * Destructor
+ */
+HottailRateTermLowZ::~HottailRateTermLowZ(){
+    Deallocate();
+    gsl_integration_workspace_free(w);
+}           
+
+/**
+ * Called after the grid is rebuilt; (re)allocates memory
+ * for all quantities
+ */
+bool HottailRateTermLowZ::GridRebuilt(){
+    this->HottailRateTerm::GridRebuilt();
+
+    Deallocate();
+            
+    this->params = new ParamStruct();
+    params->rGrid = this->rGrid; 
+    params->dist = distHT;
+    
+    Epar_prev = new real_t[nr];
+    Epar = new real_t[nr]; 
+    
+    params->sp_cond = new real_t[nr];
+    params->Zeff = new real_t[nr];
+    params->ncold = new real_t[nr];
+    params->tau = new real_t[nr];
+    params->j0 = new real_t[nr];
+    params->lnL = new real_t[nr];
+    params->Ec = new real_t[nr];
+    
+    integraloff0inGamma = new real_t[nr];
+
+    return true;
+}
+
+/**
+ * Rebuilds quantities used by this equation term.
+ * Note that the equation term uses the _energy distribution_
+ * from AnalyticDistributionHottail which differs from the 
+ * full distribution function by a factor of 4*pi
+ */
+void HottailRateTermLowZ::Rebuild(const real_t t, const real_t dt, FVM::UnknownQuantityHandler* u) {
+    if (Epar == nullptr){
+    	Epar = new real_t[nr];
+    	const real_t *Einit = u->GetUnknownInitialData(id_Efield);
+    	for(len_t ir=0; ir<nr; ir++){
+    	    Epar[ir] = Einit[ir];
+    	}
+    }
+    this->dt = dt;
+    bool newTimeStep = (t!=tPrev);
+    if(newTimeStep)
+        tPrev = t;
+    
+    params->sp_cond = runawayFluid->GetElectricConductivity();
+    params->Zeff = ionHandler->GetZeff(); 
+    params->ncold = u->GetUnknownData(id_ncold);
+    params->tau = u->GetUnknownData(id_tau);
+    params->j0 = u->GetUnknownInitialData(id_johm);
+    
+    for(len_t ir=0; ir<nr; ir++){
+    
+        params->ir = ir;
+        params->lnL[ir] = lnL->evaluateAtP(ir,0); 
+        params->Ec[ir] = 4*M_PI*(params->ncold[ir])*(params->lnL[ir])*Constants::r0*Constants::r0*Constants::c*Constants::c*Constants::me / Constants::ec; 
+        
+        if(newTimeStep)
+            Epar_prev[ir] = Epar[ir];
+        Epar[ir] = evaluate_Epar(params);
+        real_t dotEpar = (Epar[ir] - Epar_prev[ir]) / dt;
+         if (dotEpar < 0){ // ensure non-negative runaway rate
+            dotEpar = 0;
+            }
```

Comment:
Is this check required? I thought the electric field was defined such that its time derivative would never be negative?